### ISSUE TITLE
[expo-updates] proguard rules error on android release

### ISSUE THREAD

**Issue by Alice-space** [expo/expo#18002]
### Summary

need add following line in proguard file
```
-keep class com.facebook.react.devsupport.**{*;}
```
or it will produce runtime error
```
2022-06-28 11:29:07.696 10905-10905/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: net.treehollow.bosonapp, PID: 10905
    java.lang.RuntimeException: Unable to create application xxx.MainApplication: java.lang.NoSuchFieldException: No field mDefaultNativeModuleCallExceptionHandler in class Lcom/facebook/react/devsupport/d; (declaration of 'com.facebook.react.devsupport.d' appears in /data/app/xxx-vTAKMm-0dGZZthNBPp8JSA==/base.apk)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7608)
        at android.app.ActivityThread.access$2600(ActivityThread.java:260)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2435)
        at android.os.Handler.dispatchMessage(Handler.java:110)
        at android.os.Looper.loop(Looper.java:219)
        at android.app.ActivityThread.main(ActivityThread.java:8668)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1109)
     Caused by: java.lang.NoSuchFieldException: No field mDefaultNativeModuleCallExceptionHandler in class Lcom/facebook/react/devsupport/d; (declaration of 'com.facebook.react.devsupport.d' appears in /data/app/xxx-vTAKMm-0dGZZthNBPp8JSA==/base.apk)
        at java.lang.Class.getDeclaredField(Native Method)
        at i9.d.l(Unknown Source:36)
        at i9.d.n(Unknown Source:8)
        at c9.c.F(Unknown Source:20)
        at expo.modules.updates.UpdatesPackage$b.d(Unknown Source:21)
        at a7.f.createReactInstanceManager(Unknown Source:71)
        at com.facebook.react.t.a(Unknown Source:9)
        at net.treehollow.bosonapp.MainApplication.onCreate(Unknown Source:13)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1202)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7588)
        at android.app.ActivityThread.access$2600(ActivityThread.java:260) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2435) 
        at android.os.Handler.dispatchMessage(Handler.java:110) 
        at android.os.Looper.loop(Looper.java:219) 
        at android.app.ActivityThread.main(ActivityThread.java:8668) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1109) 
```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

45.0.6

### Environment

 ```
expo-env-info 1.0.4 environment info:
    System:
      OS: macOS 12.4
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.15.1 - /opt/homebrew/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 8.11.0 - /opt/homebrew/bin/npm
      Watchman: 2022.06.20.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
      Android SDK:
        API Levels: 29, 30, 31
        Build Tools: 28.0.3, 29.0.2, 30.0.2, 30.0.3, 31.0.0, 32.0.0, 32.0.0, 32.1.0, 33.0.0
        System Images: android-31 | Google Play ARM 64 v8a, android-32 | Google APIs ARM 64 v8a
        Android NDK: 22.1.7171670
    IDEs:
      Android Studio: Chipmunk 2021.2.1 Patch 1 Chipmunk 2021.2.1 Patch 1
      Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
    npmPackages:
      @expo/webpack-config: ~0.16.21 => 0.16.24 
      expo: ^45.0.6 => 45.0.6 
      react: 17.0.2 => 17.0.2 
      react-dom: 17.0.2 => 17.0.2 
      react-native: 0.68.2 => 0.68.2 
      react-native-web: 0.17.7 => 0.17.7 
    Expo Workflow: bare
```

### Reproducible demo

just turn proguard on with expo-updates enabled

**Comment by Alice-space** [comment]
I use managed workflow to produce bare project every build time

**Comment by Alice-space** [comment]
And production releases should not depend on devsupport ?

**Comment by monkeyK1n9** [comment]
I have the same problem but I am on managed workflow. Any solution?

**Comment by raajnadar** [comment]
The PR is on its way https://github.com/expo/expo/pull/18035

**Comment by esamelson** [comment]
Hey all -- thanks for the report, this is now fixed in expo-updates@0.13.3.

> And production releases should not depend on devsupport ?

You're right in theory, but the `devsupport` package in RN does currently include some classes that are used in all release builds -- such as the `DisabledDevSupportManager` class. It's just a consequence of how deeply the dev support stuff is integrated into RN, especially on Android 🤷

**Comment by raajnadar** [comment]
Do we still need to add the proguard snippet for bare apps?
